src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMapping.java

```diff
@@ -0,0 +1,133 @@
+
+        if (nonNull(overrideFields.getDuration()) && overrideFields.getDuration().intValue() >= MIN_HEARING_DURATION) {
+            return overrideFields.getDuration().intValue();
+        }
+
+        Integer duration = getHearingDurationAdjournment(caseData);
+        if (isNull(duration)) {
+            duration = getHearingDurationBenefitIssueCodes(caseData, referenceDataServiceHolder);
+        }
+
+        return nonNull(duration) ? duration : DURATION_DEFAULT;
+    }
+
+    public static Integer getHearingDurationAdjournment(SscsCaseData caseData) {
+
+        if (DURATION_TYPE_NON_STANDARD_TIME_SLOT.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationType())
+            && isNotBlank(caseData.getAdjournCaseNextHearingListingDuration())) {
+            int duration =  Integer.parseInt(caseData.getAdjournCaseNextHearingListingDuration());
+
+            if (DURATION_UNITS_SESSIONS.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationUnits())
+                && duration >= MIN_HEARING_SESSION_DURATION) {
+                return duration * DURATION_SESSIONS_MULTIPLIER;
+            } else if (DURATION_UNITS_MINUTES.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationUnits())
+                && duration >= MIN_HEARING_DURATION) {
+
+
+                return duration;
+            }
+        }
+
+        return null;
+    }
+
+    public static Integer getHearingDurationBenefitIssueCodes(SscsCaseData caseData, ReferenceDataServiceHolder referenceDataServiceHolder) {
+        uk.gov.hmcts.reform.sscs.reference.data.model.HearingDuration hearingDuration = referenceDataServiceHolder.getHearingDurations().getHearingDuration(
+            caseData.getBenefitCode(), caseData.getIssueCode());
+
+        if (isNull(hearingDuration)) {
+            return null;
+        }
+
+        if (isYes(caseData.getAppeal().getHearingOptions().getWantsToAttend())) {
+            Integer duration = isInterpreterRequired(caseData)
+                ? hearingDuration.getDurationInterpreter()
+                : hearingDuration.getDurationFaceToFace();
+            return referenceDataServiceHolder.getHearingDurations()
+                .addExtraTimeIfNeeded(duration, hearingDuration.getBenefitCode(), hearingDuration.getIssue(),
+                                      getElementsDisputed(caseData)
+                );
+        } else if (HearingsChannelMapping.isPaperCase(caseData)) {
+            return hearingDuration.getDurationPaper();
+        } else {
+            return null;
+        }
+    }
+
+    public static List<String> getNonStandardHearingDurationReasons() {
+        // TODO Future Work
+        return Collections.emptyList();
+    }
+
+    @SuppressWarnings({"PMD.CyclomaticComplexity", "PMD.NPathComplexity"})
+    public static List<String> getElementsDisputed(SscsCaseData caseData) {
+        List<ElementDisputed> elementDisputed = new ArrayList<>();
```

Comment:
While the multiple same if statements is clear, it does seem a little unnecessary. The repetition here could be improved with a foreach loop. As they are all List<ElementDisputed>, you could make a temp list of the nine caseData.getElementsDisputedGeneral() etc. in place of each of the below if statements. Then foreach through that and you would only need a single `if (isNotEmpty(el))`

Reply:
Idiot question: what would be the type of the List of methods and would not the stanza assigning this new list be much the same repetitive blob of code as this is already? I moved this block from the HearingDetailsMapping file as it is used in durations more than details.

Reply:
I was thinking something like this
```
List<List<ElementDisputed>> elementsToCheck = List.of(
  caseData.getElementsDisputedGeneral(),
  caseData.getElementsDisputedSanctions(),
  ...
);
```
Then
```
elementsToCheck.forEach(list -> 
  if (isNotEmpty(list)) elementDisputed.addAll(list);
);
```

Reply:
So the temp list is a list of lists not a list of methods., I will give it a go and see what it looks like.